nemo_curator/utils/semdedup_utils.py

```diff
@@ -412,18 +417,20 @@ def extract_pruned_data(
         results_df = results_df.persist()
         progress(results_df)
 
-        results_df.to_parquet(output_parquet_path)
+        try:
+            results_df.to_parquet(output_parquet_path)
+        except TypeError:
+            # Catching "Implicit conversion to a host PyArrow object via __arrow_array__ is not allowed"
+            logger.info("No semantic duplicates found")
+            return total_records, 0, total_records
```

Comment:
I use this to handle cases with no duplicates. Unfortunately, I tried a lot of ways to check the DataFrame, but none of them were working for me. A try/catch works, though (see newly added `test_no_sem_dedup` as an example).